### PR TITLE
Add Doc Example test cases for FileBasedSystemAccessControl

### PR DESCRIPTION
**Pull request by haldes**
There was a issue with my fork so i have to close [PR-8830](https://github.com/trinodb/trino/pull/8830)
and re-create it in this PR.

I have made all the changes discussed in [PR-8830](https://github.com/trinodb/trino/pull/8830)

@kokosing @findepi @dain  @mosabua 

Follow up from this discussion [3952](https://github.com/trinodb/trino/issues/3952)

I have added the test cases for the example docs for FileBasedSystemAccessControl.

Catalog Rules
Schema Rules
Table Rules

### CODE REVIEW

**Comment by haldes** [comment]
@kokosing @findepi @dain @mosabua
Gentle Reminder !!! Any thing else needs to be done?

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -760,6 +765,107 @@ public void testQueryNotSet()
...
+    @Test
+    public void testCatalogRulesDocsExample()
+    {
+        String rulesFile = new File("../../docs/src/main/sphinx/security/catalog_rules.json").getAbsolutePath();
+        SystemAccessControl accessControlManager = newFileBasedSystemAccessControl(ImmutableMap.of("security.config-file", rulesFile));
+
+        // admin to access the mysql and the system catalog
+        accessControlManager.checkCanAccessCatalog(new SystemSecurityContext(admin, queryId), "mysql");
+        accessControlManager.checkCanAccessCatalog(new SystemSecurityContext(admin, queryId), "system");
+        accessControlManager.checkCanAccessCatalog(new SystemSecurityContext(admin, queryId), "hive");
```
Comment (hashhar) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:777]: nit: I'd recommend to add all the hive checks together at the end - it's easier to see that way that `hive` is allowed to all identities.

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 2]
```diff
@@ -70,6 +70,11 @@
     private static final CatalogSchemaTableName aliceView = new CatalogSchemaTableName("alice-catalog", "schema", "view");
     private static final Optional<QueryId> queryId = Optional.empty();
 
+    private static final Identity peppa = Identity.forUser("peppa").withGroups(ImmutableSet.of("finance")).build();
```
Comment (hashhar) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:73]: nit: maybe something like `finance_user` and `human_resources_user` would make the tests easier to read without having to refer to how the identity is created?

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 3]
```diff
@@ -760,6 +765,107 @@ public void testQueryNotSet()
...
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessage("Access Denied: Cannot rename schema from postgres.some_schema to some_new_schema");
+        assertThatThrownBy(() -> accessControlManager.checkCanShowCreateSchema(new SystemSecurityContext(guest, queryId), new CatalogSchemaName("postgres", "some_schema")))
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessage("Access Denied: Cannot show create schema for postgres.some_schema");
+
+        // treat all users as owners of the default.default schema
+        accessControlManager.checkCanCreateSchema(new SystemSecurityContext(alice, queryId), new CatalogSchemaName("default", "default"));
+        accessControlManager.checkCanDropSchema(new SystemSecurityContext(alice, queryId), new CatalogSchemaName("default", "default"));
+        accessControlManager.checkCanShowCreateSchema(new SystemSecurityContext(alice, queryId), new CatalogSchemaName("default", "default"));
```
Comment (hashhar) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:824]: Maybe also test what happens with `checkCanRenameSchema`?

**Review by hashhar — COMMENTED** [review]
Looks mostly good to me % comments.

I'll defer to @kokosing since he's more familiar with access controls.

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java (hunk 2 above)
Comment (kokosing) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:73]: What is the semantic of this user? I don't see it defined in above files.

**Review by kokosing — APPROVED** [review]

**Comment by haldes** [comment]
Thanks @hashhar @kokosing for reviewing
Sorry for replying late.
Looks like there are more changes from [Add role to file based system access control](https://github.com/trinodb/trino/commit/868fed9990c5be6d61376f802f09185babbb6f64) post my commit.
Will resolve the changes and commit back.

**Comment by colebow** [comment]
:wave: @haldes - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**